content/docs/installation/features.md

```diff
@@ -0,0 +1,62 @@
+---
+title: Features
+description: cert-manager feature policy
+---
+
+New cert-manager features and functionality is often initially implemented behind a feature gate. This is so as to not break users with functionality that has not yet been tested in production as well as to give us a chance to remove or modify API fields and functionality following user feedback.
+
+We have alpha and beta features. We do not aim to keep any of the features in alpha or beta stage indefinitely. Feature gating should only be used for functionality that can eventually be turned on by default for all users (GA).
+
+A feature gate can be toggled on/off using `--feature-gates` flags on cert-manager controller. For feature gated functionality that comes with new API fields there is also a corresponding feature gate on webhook that also needs to be enabled using a `--feature-gates` flag if you want to use it.
+
+**Alpha**
+
+All alpha features are off by default.
+We keep the right to remove or change alpha features without a warning.
+An API field that is part of an alpha feature and requires a webhook feature flag to be used also can be removed from the API without a warning.
+An alpha feature might not work for all versions of Kubernetes supported by cert-manager.
+
+**Beta**
+
+All beta features are off by default.
+Beta features will not be removed, but can be changed.
+If the feature gets changed in incompatible ways, we will provide migration instructions.
+A beta feature will work with all versions of Kubernetes supported by cert-manager.
+
+**GA**
+
+A feature that is GA is on by default and cannot be disabled.
+With regards to API fields and their functionality, we keep Kubernetes API compatibility promise https://cert-manager.io/docs/installation/api-compatibility/.
+The feature flag for a GA feature will eventually be removed.
```

Comment:
suggestion: I'm not sure if there'd be any reason to ever _actually_ remove one - it feels like a needless way to break people.

It doesn't seem like a huge burden to move old feature flags to `OldFeatureFlags []string`, where if any of them are passed we just log a warning and then move on.

Removing would make sense if we were ever gonna re-use a flag name but I think that could get confusing tbh

Reply:
Yeah I agree with you, I think removing would make sense if we ended up with a huge number of features gates, so that the help output would become unreadable, but this is currently not the case.

Reply:
The downside would be that if it's a feature gated opt-in functionality that gets enabled via flag in GA, it could be confusing that it appears in two places. But since this would be non-functional, we could change it if folks complained.